frontend/app/(tabs)/home.tsx

```diff
@@ -28,18 +39,55 @@ const Index = () => {
   };
 
   const loadMoreCards = useCallback(async () => {
-    if (isLoading) return;
+    if (isLoading || searchText !== '') return;
     try {
       setIsLoading(true);
       const newCards = await fetchCards(page + 1, CARDS_PER_PAGE);
-      setCards((prev) => [...prev, ...newCards]);
+      const updatedCards = [...cards, ...newCards];
+      setCards(updatedCards);
+      setFilteredCards(updatedCards);
       setPage((prev) => prev + 1);
     } catch (error) {
       console.error('Error loading more cards:', error);
     } finally {
       setIsLoading(false);
     }
-  }, [page, isLoading]);
+  }, [page, isLoading, cards, searchText]);
+
+  const handleSearch = (text: string) => {
+    setSearchText(text);
+    if (text.trim() === '') {
+      setFilteredCards(cards);
+    } else {
+      const filtered = cards.filter(
+        (card) =>
+          card.title.toLowerCase().includes(text.toLowerCase()) ||
+          (card.content &&
+            card.content.toLowerCase().includes(text.toLowerCase())),
```

Comment:
We will need to change this when we add examples and descriptions, we might want the search to happen in the backend.